sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -339,7 +339,8 @@ case class DataSource(
         val baseRelation =
           dataSource.createRelation(sparkSession.sqlContext, caseInsensitiveOptions)
         if (baseRelation.schema != schema) {
-          throw new AnalysisException(s"$className does not allow user-specified schemas.")
+          throw new AnalysisException(s"$className does not allow user-specified schemas, " +
+              s"source schema: ${baseRelation.schema}, user-specific schema: ${schema}")
```

Comment:
I would print out `schema.catalogString` for better string representation.

Reply:
THX for your replies,  I will fix it later.

Reply:
I think `schema.catalogString` is still not better than `schema.treeString` because `schema.catalogString` cannot express `nullable` values. What do you think?

Reply:
Ideally we should print out the problematic field then during comparison.
Spark has been using `catalogString` for now for simplicity. Let's stick to `catalogString` for now to be consistent.